micrometer-observation/src/main/java/io/micrometer/observation/Observation.java

```diff
@@ -1294,6 +1307,15 @@ interface ContextView {
          */
         <T> T getOrDefault(Object key, T defaultObject);
 
+        /**
+         * Returns an element or default if not present.
+         * @param key key
+         * @param defaultObjectSupplier supplier for default object to return
+         * @param <T> value type
+         * @return object or default if not present
+         */
+        <T> T getOrDefault(Object key, Supplier<T> defaultObjectSupplier);
```

Comment:
I guess that should be a default method of we don't want to break current users (assuming that anyone has ever created their own implementation of ObservationRegistry)

Reply:
@marcingrzejszczak Thanks for the feedback! I updated this based on your feedback.